ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteImporter.java

```diff
@@ -867,76 +864,84 @@ private void mergeXnodeDanglingLines(UcteNetwork ucteNetwork, Network network) {
-                        .setR(dl1.getR())
-                        .setX(dl1.getX())
-                        .setG1(dl1.getG())
-                        .setG2(0.0)
-                        .setB1(dl1.getB())
-                        .setB2(0.0)
-                        .setXnodeP(xnodeP1)
-                        .setXnodeQ(xnodeQ1)
-                        .line2()
-                        .setId(dl2.getId())
-                        .setR(dl2.getR())
-                        .setX(dl2.getX())
-                        .setG1(0.0)
-                        .setG2(dl2.getG())
-                        .setB1(0.0)
-                        .setB2(dl2.getB())
-                        .setXnodeP(xnodeP2)
-                        .setXnodeQ(xnodeQ2)
-                        .setUcteXnodeCode(xnodeCode)
-                        .add();
+                boolean switchDanglingLinesOrder = dlToProcess.getId().compareTo(dlMatchingDlToProcess.getId()) > 0;
+                DanglingLine dlAtSideOne = switchDanglingLinesOrder ? dlMatchingDlToProcess : dlToProcess;
+                DanglingLine dlAtSideTwo = switchDanglingLinesOrder ? dlToProcess : dlMatchingDlToProcess;
 
-                addElementNameProperty(mergeLine, dl1, dl2);
-                addGeographicalNameProperty(ucteNetwork, mergeLine, dl1, dl2);
+                createTieLine(ucteNetwork, network, dlAtSideOne, dlAtSideTwo);
 
-                if (dl1.getCurrentLimits() != null) {
-                    mergeLine.newCurrentLimits1()
-                            .setPermanentLimit(dl1.getCurrentLimits().getPermanentLimit()).add();
-                }
-                if (dl2.getCurrentLimits() != null) {
-                    mergeLine.newCurrentLimits2()
-                            .setPermanentLimit(dl2.getCurrentLimits().getPermanentLimit()).add();
-                }
+                dlToProcess.remove();
+                dlMatchingDlToProcess.remove();
 
-                mergeLine.addExtension(MergedXnode.class, new MergedXnode(mergeLine, rdp, xdp, xnodeP1, xnodeQ1, xnodeP2, xnodeQ2,
-                        dl1.getId(), dl2.getId(), xnodeCode));
+                danglingLinesToProcess.remove(dlMatchingDlToProcess);
+            }
+            danglingLinesToProcess.remove(dlToProcess);
+        }
+    }
+
+    private void createTieLine(UcteNetwork ucteNetwork, Network network, DanglingLine dlAtSideOne, DanglingLine dlAtSideTwo) {
+        // lexical sort to always end up with same merge line id
+        String mergeLineId = dlAtSideOne.getId() + " + " + dlAtSideTwo.getId();
+
+        // create XNODE merge extension
+        float rdp = (float) (dlAtSideOne.getR() / (dlAtSideOne.getR() + dlAtSideTwo.getR()));
+        float xdp = (float) (dlAtSideOne.getX() / (dlAtSideOne.getX() + dlAtSideTwo.getX()));
+        double xnodeP1 = dlAtSideOne.getP0();
+        double xnodeQ1 = dlAtSideOne.getQ0();
+        double xnodeP2 = dlAtSideTwo.getP0();
+        double xnodeQ2 = dlAtSideTwo.getQ0();
```

Comment:
Maybe you should respect your own style by removing this variable used once and call `dlAtSideOne.getP0()` (and the others) in the adder.

Reply:
They are used twice (also in the merged Xnode extension). Do I really change it ?

Reply:
I didn't notice that these variables were used twice. Keep it like this